core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -778,8 +778,17 @@ private static synchronized List<alluxio.grpc.Metric> reportMetrics(InstanceType
               entry.getKey(), gauge.getValue().getClass().getSimpleName());
           continue;
         }
-        rpcMetrics.add(Metric.from(entry.getKey(),
-            ((Number) gauge.getValue()).longValue(), MetricType.GAUGE).toProto());
+        // As the value is a number, it can also be reported to master as "COUNTER" type.
+        // Then it can also be aggregated in the same way as "COUNTER" at master side.
+        long value = ((Number) gauge.getValue()).longValue();
+        Long prev = LAST_REPORTED_METRICS.replace(entry.getKey(), value);
+        if (prev == null) {
+          LAST_REPORTED_METRICS.put(entry.getKey(), value);
+        }
+        double diff = prev != null ? value - prev : value;
+        if (diff != 0) { // Only report non-zero counter values
+          rpcMetrics.add(Metric.from(entry.getKey(), diff, MetricType.COUNTER).toProto());
```

Comment:
why do we report the diff instead of the absolute value?

Reply:
Context at the server side: https://github.com/Alluxio/alluxio/blob/a4ecde4b26c9661931bc860ca708ce8ec280ca39/core/server/master/src/main/java/alluxio/master/metrics/MetricsStore.java#L126-L129

Also a question for @LuQQiu : why are only counters aggregated at the master, metrics of the other types are ignored?

Reply:
Previously only counters are send to master, and we are not quite sure how to aggregate the other values, but it make sense if other types of metrics are aggregatable and aggregate them in other ways